src/lib/protocols/ssh.c

```diff
@@ -60,17 +62,102 @@ static void ndpi_search_ssh_tcp(struct ndpi_detection_module_struct *ndpi_struct
   
 /* ************************************************************************ */
 
+static int ssh_has_old_signature(char *signature) {
+  int is_old = 0;
+  int i = 0;
+
+  char *old_versions[46] = {
```

Comment:
This is a linear scan that could be avoided if you could parse the version as numeric field instead of treating the version as a string. In essence anything other than OpenSSH 7.x looks obsolete.